test/e2e_node/pod_resize_test.go

```diff
@@ -307,56 +309,164 @@ func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
-			expectedMemLimitInBytes := tc.Resources.Limits.Memory().Value()
-			cpuRequest := tc.Resources.Requests.Cpu()
-			cpuLimit := tc.Resources.Limits.Cpu()
-			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
-				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
-			} else {
-				expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
-			}
-			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
-			if cpuLimit.IsZero() {
-				cpuQuota = -1
+		var cpuRequest, cpuLimit *resource.Quantity
+		var expectedMemLimitInBytes int64
+
+		accumulateContainerValues := func() {
+			// Accumulate container resources for verifying pod
+			if cpuRequest != nil {
+				podCPURequestMilliValue += cpuRequest.MilliValue()
 			}
-			expectedCPULimitString = strconv.FormatInt(cpuQuota, 10)
-			expectedMemLimitString = strconv.FormatInt(expectedMemLimitInBytes, 10)
-			if podOnCgroupv2Node {
-				if expectedCPULimitString == "-1" {
-					expectedCPULimitString = "max"
+			if podCPULimitMilliValue >= 0 {
+				if cpuLimit == nil || cpuLimit.IsZero() {
+					podCPULimitMilliValue = -1
+				} else {
+					podCPULimitMilliValue += cpuLimit.MilliValue()
 				}
-				expectedCPULimitString = fmt.Sprintf("%s %s", expectedCPULimitString, CPUPeriod)
-				if expectedMemLimitString == "0" {
-					expectedMemLimitString = "max"
-				}
-				// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
-				// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
-				expectedCPUShares = int64(1 + ((expectedCPUShares-2)*9999)/262142)
 			}
-			if expectedMemLimitString != "0" {
-				err := verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString)
-				if err != nil {
-					return err
+			if podMemoryLimitInBytes >= 0 {
+				if expectedMemLimitInBytes == 0 {
+					podMemoryLimitInBytes = -1
+				} else {
+					podMemoryLimitInBytes += expectedMemLimitInBytes
 				}
 			}
-			err := verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString)
-			if err != nil {
-				return err
+		}
+
+		if ci.Resources == nil {
+			accumulateContainerValues()
+			continue
+		}
+
+		tc, _ := makeTestContainer(ci)
+		if tc.Resources.Limits == nil && tc.Resources.Requests == nil {
+			accumulateContainerValues()
+			continue
+		}
+
+		var expectedCPUShares int64
+		var expectedCPULimitString, expectedMemLimitString string
+		expectedMemLimitInBytes = tc.Resources.Limits.Memory().Value()
+		cpuRequest = tc.Resources.Requests.Cpu()
+		cpuLimit = tc.Resources.Limits.Cpu()
+		if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+			expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+		} else {
+			expectedCPUShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+		}
+		cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+		if cpuLimit.IsZero() {
+			cpuQuota = -1
+		}
```

Comment:
nit: to reduce the function call when unnecessary

```suggestion
		cpuQuota := -1
		if !cpuLimit.IsZero() {
			cpuQuota = kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
		}
```

Reply:
Done. Thanks.